### PR TITLE
read gtfs config from json file

### PR DESCRIPTION
**Pull request by nberard**
json file expected should look like:

```json
{
  "contributor": {
    "contributor_id": "OIF",
    "contributor_name": "OIF - STIF Ile de France",
    "contributor_website": "http://www.vianavigo.com",
    "contributor_license": "ODbL"
  },
  "dataset": {
    "contributor_id": "OIF",
    "dataset_id": "OIF:0",
    "dataset_start_date": "20170607",
    "dataset_end_date": "20170702",
    "dataset_type": "0",
    "dataset_desc": "NIO_auto_20170618",
    "dataset_system": "GTFS V2 Ligne",
    "dataset_extrapolation": 0
  }
}
```

to test:
```
cargo run -- -i /path/to/input -o /path/to/output -c /path/to/config
```

### CODE REVIEW

File: src/gtfs/mod.rs [hunk 1]
```diff
@@ -18,10 +18,24 @@ mod read;
...
     let mut collections = Collections::default();
+
+    if let Some(config_path) = config_path {
+        let json_config_file = File::open(config_path).unwrap();
+        let config: Config = serde_json::from_reader(json_config_file).unwrap();
+        info!("config loaded: {:#?}", config);
+        let (contributors, datasets) = read::read_config(config);
+        collections.contributors = contributors;
+        collections.datasets = datasets;
+    }
```
Comment (TeXitoi) [inline src/gtfs/mod.rs]: We don't create a default one if that's not given?

Comment (datanel) [reply src/gtfs/mod.rs]: @TeXitoi  I don't think so

Comment (datanel) [reply src/gtfs/mod.rs]: contributors.txt and datasets.txt are created but empty

File: src/gtfs/mod.rs [hunk 2]
```diff
@@ -18,10 +18,24 @@ mod read;
...
 use std::path;
 use {Collections, PtObjects};
+use std::fs::File;
+use gtfs::read::Config;
 
-pub fn read<P: AsRef<path::Path>>(path: P) -> PtObjects {
-    let path = path.as_ref();
+extern crate serde_json;
+
+pub fn read<P: AsRef<path::Path>>(path: P, config_path: Option<P>) -> PtObjects {
```
Comment (datanel) [inline src/gtfs/mod.rs]: we should instead now return a `Result<PtObjects>` (see https://github.com/CanalTP/navitia_model/blob/master/src/lib.rs#L48)

File: src/gtfs/mod.rs [hunk 3]
```diff
@@ -18,10 +18,24 @@ mod read;
...
     let mut collections = Collections::default();
+
+    if let Some(config_path) = config_path {
+        let json_config_file = File::open(config_path).unwrap();
```
Comment (datanel) [inline src/gtfs/mod.rs]: replace .unwrap by `?`

File: src/gtfs/mod.rs [hunk 4]
```diff
@@ -18,10 +18,24 @@ mod read;
...
+        let config: Config = serde_json::from_reader(json_config_file).unwrap();
```
Comment (datanel) [inline src/gtfs/mod.rs]: replace .unwrap by `?`

File: src/gtfs/mod.rs [hunk 5]
```diff
@@ -18,10 +18,24 @@ mod read;
...
+        info!("config loaded: {:#?}", config);
+        let (contributors, datasets) = read::read_config(config);
```
Comment (datanel) [inline src/gtfs/mod.rs]: read_config should return a Result I think

File: examples/gtfs_reader.rs [hunk 6]
```diff
@@ -20,7 +20,7 @@ extern crate navitia_model;
 extern crate serde_json;
 
 fn main() {
-    let objects = navitia_model::gtfs::read(".");
+    let objects = navitia_model::gtfs::read(".", None);
```
Comment (datanel) [inline examples/gtfs_reader.rs]: you can do the same as `example/test.rs` with `run` function

And we should do the same in `bin/gtfs2ntfs.rs`

**Pull request merged**